src/figuresClasses.ts

```diff
@@ -1,19 +1,91 @@
-export interface Figure {
+enum Figures {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+enum Colors {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
 
+export interface Figure {
+  shape: Figures,
+  color: Colors,
+  getArea():number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Figures = Figures.Triangle;
+
+  constructor(
+    public color: Colors,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Invalid side value of triangle');
+    }
 
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error('Sides cannot form a triangle');
+    }
+  }
+
+  getArea():number {
+    const p:number = (this.a + this.b + this.c) / 2;
+    const triangleArea = Math.sqrt(p
+      * (p - this.a) * (p - this.b) * (p - this.c));
```

Comment:
quite difficult to read. Consider creating some extra variables or moving some logic to a separate helper

Reply:
Done.